src/server/worker/datum/datum.go

```diff
@@ -235,6 +240,24 @@ func (d *Datum) withData(cb func() error) (retErr error) {
 	})
 }
 
+func (d *Datum) createEnvFile() (retErr error) {
+	var envStr string
+	for _, e := range d.env {
+		envStr += e + "\n"
+	}
+	f, err := os.Create(path.Join(d.PFSStorageRoot(), common.EnvFileName))
```

Comment:
You can golf this down to `os.WriteFile(filepath.Join(...), []byte(envStr), <permissions>)`.  Also, `path.Join` is for URLs, and `filepath.Join` is for the OS that the code is running on.  Obviously we don't run on an OS that can't take slash-separated filenames, but I think `filepath` is technically more correct.

Reply:
I'm going to leave it as `path.Join` for now to be consistent with the rest of the code, but I added a TODO for this.